### DISCUSSION TITLE
6591 - image2inmemory nicht funktionsfähig?

### DISCUSSION THREAD

**Discussion by Bodenseematze** [Freetz-NG/freetz-ng#328] [push firmware]
Hallo,

ich hatte jetzt das erste Mal eine Kabelbox unter meinen Fittichen.
Natürlich wollte ich auch die mit freetz-ng bestücken - also wie üblich:
 1.) einigermaßen minimale Konfiguration auswählen
 2.) Image bauen
 3.) Image für Erst-Bestückung zum In-Memory-Image umwandeln
 
Die ersten beiden Schritte haben eigentlich geklappt - wobei mir die Größe des Ergebnisse "spanisch" vorkommt - 57 MB, und das für nahezu minimal?
Dann wollte ich mit dem dritten Schritt über das Script "tools/image2inmemory" das Image umwandeln.
Da kommt aber nur eine Fehlermeldung, das das Image nicht ausgepackt werden könne:
````
> tools/image2inmemory
Reading /home/freetz/workdir/images/6591_07.27.ger_freetz-ng-18412M-84a329449_20210615-132601.image

ERROR: Unpacking '/home/freetz/workdir/images/6591_07.27.ger_freetz-ng-18412M-84a329449_20210615-132601.image' failed.
````

So jetzt meine Fragen:
  Wieso ist das Image trotz Mini-Konfiguration so groß?
  Wieso funktioniert das image2inmemory-Tool nicht? Mache ich was falsch oder ist das nicht für Kabelboxen geeignet?
Ist die übliche Vorgehensweise zum Freetzen für Kabelboxen eine andere wie für die "normalen" DSL-Boxen?

[6591_07.27_2021-06-15_config.txt](https://github.com/Freetz-NG/freetz-ng/files/6656065/6591_07.27_2021-06-15_config.txt)

[6591_07.27_2021-06-15_config_compressed.txt](https://github.com/Freetz-NG/freetz-ng/files/6656068/6591_07.27_2021-06-15_config_compressed.txt)

**Comment by fda77** [comment]
Die Datei von AVM ist 57282560 Bytes, was soll daran komisch sein dass die mir Freetz "57mb" ist? Das hängt davon ab was du entfernst+hinzufügst. Entfern halt mehr wenn du ein kleineres Image haben willst

Das mit diesem blöden inmemory ist und war schon immer Mist. Es ist ein Zwischenformat während des Flashvorgang, sonst nichts. Ein gutes Flashprogramm/script macht das automatisch FALLS NÖTIG. Dies ist nur bei 1 der aktuell 5 Flashing-Arten bei AVM der Fall.
Nimm einfach push-firmware !!!!!!111
Falls du Windows zum flashen verwenden willst musst du dir die ftp-Befehle zusammensuchen

-> 0b470ad99

**Comment by PeterPawn** [comment]
Seit wann macht das bei einer Puma7-Box überhaupt Sinn? Wo wird das Format denn da verwendet? Wie bringt man denn die ZWEI Kernel und die ZWEI Dateisysteme in diesem Format unter? Erst beide Kernel oder beide gemischt? Es gibt zwar vermutlich(!) auch bei diesem Modell die Option, ein System aus dem RAM zu starten ... aber wenn das jemand außerhalb von AVM konkret und mit den notwendigen Einzelheiten wissen sollte, hat er es (meines Wissens) bisher immer schön für sich behalten.

@fda77:
Die Ansichten, was ein "gutes Tool" so machen sollte, gehen offensichtlich deutlich auseinander, aber nicht nur zwischen Dir und mir, sondern auch zwischen Dir und einer größeren "Community": https://de.wikipedia.org/wiki/Unix-Philosophie

**Reply by Bodenseematze** [reply]
Es war mir nicht bewußt, dass das inmemory-Format eigentlich nicht notwendig ist?!
Ich glaube mich erinnern zu können, dass ich die gepackten Images bisher nicht per EVA-FTP-Client booten konnte und flashen über die AVM-Oberfläche ging auch nicht (beim ersten Mal) - erst nachdem ich schon eijn Image erstmalig draufgeschoben hatte.

Und die o.a. Box scheint ein Vodafone-Branding gehabt zu haben, zumindest gab es die Option zum Hochladen eines Images aus einer Datei dort nicht ("gab" deshalb, weil die inzwischen wieder verkauft wurde wg. massivem Ärger mit einem neuen Vodafone Kabelanschluss und der Unfähigkeit der Vodafone-Mitarbeiter, die o.a. FritzBox als "eigene" Box an Kabel zu hängen).

Achja, und push-firmware hat bei mir bisher eigentlich auch noch nie funktioniert, das sucht immer die Box und findet sie nicht - liegt wahrscheinlich daran, dass es sich beim dem verwendeten Linux-Unterbau um eine VM handelt...

Für mich sollte ein "gutes Tool" v.a. aussagekräftige Fehlermeldungen ausspucken und nicht nur ein "fehlgeschlagen" - mir fehlt da ein warum es fehlgeschlagen ist bzw. in dem Fall, dass es nicht möglich/notwendig ist... ;-)
 
@PeterPawn: wenn ich Dich richtig verstehe, geht die Methode mit dem Booten aus dem RAM also bei den Kabelboxen generell nicht (bzw. es ist nicht bekannt, wie es geht)?

**Comment by PeterPawn** [comment]
> wenn ich Dich richtig verstehe

Exakt. Wenn Du eine 6591 mit eigener Firmware flashen möchtest, solltest Du Dir (vermutlich unter Deinem Alter-Ego [@Massa](https://www.ip-phone-forum.de/members/massa.60474/)?) den betreffenden Thread im IPPF (da gibt's eigentlich für dieses Modell tatsächlich nur den einen, zumindest wenn es nach Relevanz geht) ansehen.

**Comment by fda77** [comment]
Wenn push firmware nicht läuft musst du die FTP Befehle halt von Hand eingeben. Mit ist sonst kein Tool/Script bekannt. @PeterPawns Windowsscript funktioniert jedenfalls gar nicht und kannst du gleich vergessen

**Reply by Bodenseematze** [reply]
Also bei mir haben Peter's Windows-Scripte in der Vergangenheit immer sehr gut funktioniert...

**Comment by PeterPawn** [comment]
@Bodenseematze:
Was soll man hierzu: https://github.com/Freetz-NG/freetz-ng/discussions/328#discussioncomment-950359 nun noch schreiben? Vielleicht hat @fda77 ja nur die falschen Tabletten genommen? Oder etwas Falsches geraucht?

Ich wüßte jedenfalls nicht, daß es tatsächlich mal - nachweislich - an meiner Windows-Version der Skript-Dateien für den Umgang mit EVA gelegen hätte (https://github.com/PeterPawn/YourFritz/blob/main/eva_tools/EVA-FTP-Client.ps1), wenn es bei einem potentiellen Anwender nicht funktionierte.

Sicherlich - es gibt ein paar Klippen, die man auch unter Windows berücksichtigen muß - aber die meisten Hürden betrafen eine ältere Implementierung des Discovery-Skripts, wo noch eine Freigabe in der Firewall erforderlich war.

Beim FTP-Zugriff existieren diese Probleme nicht (weil alles ausgehender Traffic ist aus Windows-Sicht) und letztlich hat sogar ein Dritter (@WileC) auf Basis dieser Skripte ein eigenes Wrapper-Skript erstellt: https://github.com/WileC/FreetzTheBox ... also laß Dir nicht einfach irgendwelchen Unsinn erzählen, sondern mach Dir selbst ein Bild.

Allerdings funktioniert das Flashen bei der 6591 halt etwas anders - das war offenbar @fda77 auch entfallen, denn er zieht weiter oben ja auch nur über das "in-memory"-Format her und korrigiert den vollkommen falschen Ansatz für eine 6591 gar nicht erst.

Wie man eine 6591 richtig flasht, steht im oben schon erwähnten IPPF-Thread - bis hin zu den "merkwürdigen" Namen für die oberen MTD-Partitionen, die sich durch das "Weiterzählen" nach der `9` in der ASCII-Tabelle ergeben und wie man diese Namen auch erfolgreich mit FTP-Programmen nutzen kann, die diese ansonsten speziell behandeln würden aufgrund der enthaltenen Sonderzeichen. 

Und selbstverständlich kann man auch EVA-FTP-Client.ps1 zum Flashen einer 6591 (hier geht es dann aber darum, eine Datei direkt in eine MTD-Partition zu schreiben) benutzen ... man muß es nur richtig aufrufen.

**Reply by fda77** [reply]
Benutzer müssen NICHT nicht wissen wie zu flashen ist - da push firmware dies anhand der .IMAGE Datei erkennt. Das steht auch so in der 1. Antwort hier

**Reply by PeterPawn** [reply]
> Benutzer müssen NICHT nicht wissen wie zu flashen ist

Sag ich doch auch immer ... Benutzer sollten schon wissen, wie das funktioniert (oder meinetwegen auch "NICHT nicht wissen", wenn Du das für passender hältst) - ansonsten stehen sie bei Problemen wie der Ochs vorm Tor und wissen auch nicht weiter.

**Reply by Bodenseematze** [reply]
Ich finde, dass beides möglich sein muss - etwas, bei dem reine Benutzer (erst einmal) nicht wissen müssen, wie das flashen vor sich geht - aber falls es nicht funktioniert, sollten sie m.E. von dem Tool/Script (wasauchimmer) an die Hand genommen werden und detailliert Fehler präsentiert bekommen und dann schrittweise begleitet werden, um selber herauszufinden welche Schritte ablaufen und bei welchem es schieft geht.
Nicht jeder muss und will so detailliert in die Materie eintauchen - aber um es einfacher zu machen, und bei Problemen eben nicht wie der Ochs vor'm Tor zu stehen, sollten das Tool in der Lage sein auf "detailliert" / "schrittweise" oder was auch immer umzuschalten...

**Comment by fda77** [comment]
https://learngerman.dw.com/de/müssen-oder-sollen/l-40504512/gr-40508500

**Reply by PeterPawn** [reply]
Wenn Du weiter fleißig Deine Übungen machst, landest Du früher oder später auch mal beim Konjunktiv II von `sollen`:

https://learngerman.dw.com/de/konjunktiv-ii-pr%C3%A4sens/l-38485158/gr-39125065

und dann kannst Du so ein `sollte` auch gleich viel besser verstehen und Dir wird vielleicht sogar der Unterschied zu `müssen` (ja, selbst zu `sollen`) verständlicher werden, wenn Du erst bei der Benutzung von Modalverben zur Beschreibung des Verhältnisses zwischen dem Subjekt und dem vorhandenen Hauptverb in einem Satz angekommen sein wirst.

**Reply by Conan179** [reply]
Herr gott, könnt ihr beiden euren Kindergarten wo anders ablassen? Eure gegenseitiges gebashe nervt.

**Comment by fda77** [comment]
Jo, das nervt ungemein. Ich klick mal auf unsubscribe

**Reply by Bodenseematze** [reply]
Verstehe ich jetzt nicht so ganz - Du hast doch oben mit dem Bashing angefangen, als Du geschrieben hattest, dass man PeterPawn's Windows-Scripte vergessen könne...
...und jetzt nervt Dich das selber???

Dann lass' es doch einfach - in meinen Augen seid ihr alle beide sehr, sehr wichtig für die FritzBox-/Freetz-Community und habt mir persönlich auch alle beide schon sehr gut geholfen!

Dass ihr in vielen Dingen völlg unterschiedlicher Meinung seid und irgendwie nicht miteinander könnt, mag ja sein - aber das muss doch nicht in Misskreditierung der Arbeit des jeweils anderen ausarten, oder?

...nur ein kleiner Wunsch meinerseits - oder anders gesagt: habt Euch lieb ;-)

**Comment by fda77** [comment]
Dieses sinnlose Gerede!
Reiss doch nicht alles aus dem Zusammenhang und verdreh es
Zitat:
```
Wenn push firmware nicht läuft musst du die FTP Befehle halt von Hand eingeben. Mit ist
sonst [ausser push firmware] kein Tool/Script bekannt. @PeterPawns Windowsscript
[eva-adam-push-irgendwas] funktioniert jedenfalls gar nicht und kannst du gleich vergessen
```
1. lesen
2. verstehen

Das ist doch die gleiche Leider wie dass jahrelang alle mit "ihrem" ru-kerneltool die ramboot/nand Boxen flashen wollten. Immerhin hat sich das mittlerweil (7490 gibts seit 2013 zu kaufen) herumgesprochen. Wahrscheihlich nur weil es eine Timebomb hat und eingestellt wurde 

Deshlab (nochmal): Wenn man nicht weiss welche Flashmethode man verwenden MUSS, nimmt man einfach ein Tool das automatisch die richtige nutzt - oder man gibt die PASSENDEN ftp-Befehle von Hand ein.

Das steht alles aber schon weiter oben .
